src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -803,14 +863,388 @@ def forward(
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.discriminators = nn.ModuleList(
+            [
+                STFTDiscriminator(
+                    filters=config.filters,
+                    in_channels=config.in_channels,
+                    out_channels=config.out_channels,
+                    n_fft=n_fft,
+                    hop_length=hop_length,
+                    win_length=win_length,
+                    kernel_size=config.kernel_size,
+                    stride=config.stride,
+                    dilations=config.dilations,
+                    max_filters=config.max_filters,
+                    filters_scale=config.filters_scale,
+                    normalized=config.normalized,
+                    norm=config.norm,
+                    activation=config.activation,
+                    activation_params=config.activation_params,
+                )
+                for n_fft, hop_length, win_length in zip(config.n_ffts, config.hop_lengths, config.win_lengths)
+            ]
+        )
+        self.num_discriminators = len(self.discriminators)
+
+    def forward(
+        self,
+        input_values: torch.Tensor,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, ModelOutput]:
+        """
+        Args:
+            input_values (`torch.Tensor` of shape `(batch_size, channels, sequence_length)`):
+                Input audio waveform.
+            return_dict (`bool`, *optional*):
+                Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+
+        Returns:
+            A tuple of `(logits, feature_maps)` where:
+                - logits is a list of discriminator outputs for each STFT scale
+                - feature_maps is a list of feature maps from each discriminator
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        logits = []
+        fmaps = []
+        for disc in self.discriminators:
+            logit, fmap = disc(input_values)
+            logits.append(logit)
+            fmaps.append(fmap)
+
+        if not return_dict:
+            return (logits, fmaps)
+
+        return DiscriminatorOutput(
+            logits=logits,
+            feature_maps=fmaps,
+        )
+
+    def compute_loss(self, real_audio, fake_audio):
+        """
+        Computes the discriminator and generator losses.
+
+        Args:
+            real_audio (`torch.Tensor` of shape `(batch_size, channels, sequence_length)`):
+                Real audio waveform.
+            fake_audio (`torch.Tensor` of shape `(batch_size, channels, sequence_length)`):
+                Generated/fake audio waveform.
+
+        Returns:
+            `tuple`:
+                - d_loss (`torch.Tensor`): Discriminator loss.
+                - g_adv_loss (`torch.Tensor`): Generator adversarial loss.
+                - fm_loss (`torch.Tensor`): Feature matching loss.
```

Comment:
I think it could be good to be more explicit in the naming! discriminator_loss generator_adversarial_loss, feature_matching_loss